File: src/OpenSilver.Showcase/Other/MaterialDesign_Styles_Kit/AccentColorConverter.fs
namespace MaterialDesign_Styles_Kit

open System
open System.Windows.Data
open System.Windows.Media
open System.Globalization

type AccentColorConverter() =
    interface IValueConverter with
        member this.Convert(value, targetType, parameter, language) =
            if value :? SolidColorBrush then
                let oldColor = (value :?> SolidColorBrush).Color
                let newR = if oldColor.R > byte(40) then byte (oldColor.R - byte(40)) else byte 0
                let newG = if oldColor.G > byte(40) then byte (oldColor.G - byte(40)) else byte 0
                let newB = if oldColor.B > byte(40) then byte (oldColor.B - byte(40)) else byte 0
                let newColor = Color.FromArgb(oldColor.A, newR, newG, newB)
                new SolidColorBrush(newColor) :> obj
            else
                value

        member this.ConvertBack(value, targetType, parameter, language) =
            raise (NotImplementedException())

File: src/OpenSilver.Showcase/Other/MaterialDesign_Styles_Kit/AccentColorConverter.cs
using System;
using System.Windows.Data;
using System.Windows.Media;
using System.Globalization;

namespace MaterialDesign_Styles_Kit
{
    public class AccentColorConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (value is SolidColorBrush)
            {
                Color oldColor = ((SolidColorBrush)value).Color;
                Byte newR = oldColor.R > 40 ? (Byte)(oldColor.R - 40) : (Byte)0;
                Byte newG = oldColor.G > 40 ? (Byte)(oldColor.G - 40) : (Byte)0;
                Byte newB = oldColor.B > 40 ? (Byte)(oldColor.B - 40) : (Byte)0;
                Color newColor = Color.FromArgb(oldColor.A, newR, newG, newB);
                return new SolidColorBrush(newColor);
            }
            return value;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}
